pkg/api/validate/dynamic/dynamic.go

```diff
@@ -69,6 +72,7 @@ type dynamic struct {
 	resourceSkusClient compute.ResourceSkusClient
 	spComputeUsage     compute.UsageClient
 	spNetworkUsage     network.UsageClient
+	getToken           func(ctx context.Context, log *logrus.Entry, clientID, clientSecret, tenantID string, aadEndpoint, resource string) (*adal.ServicePrincipalToken, error)
 }
 
 type AuthorizerType string
```

Comment:
What is the purpose of this ? It should be an interface (getToken) not a function.

Reply:
Thanks for pointing this out. I've update the packages to use interfaces rather than a function.